src/main/java/com/synopsys/integration/detect/lifecycle/autonomous/ScanTypeDecider.java

```diff
@@ -0,0 +1,181 @@
+package com.synopsys.integration.detect.lifecycle.autonomous;
+
+import com.synopsys.integration.configuration.property.types.enumallnone.list.AllNoneEnumCollection;
+import com.synopsys.integration.detect.configuration.DetectProperties;
+import com.synopsys.integration.detect.configuration.DetectPropertyConfiguration;
+import com.synopsys.integration.detect.configuration.enumeration.DetectTool;
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.FileVisitResult;
+import java.nio.file.FileVisitor;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import org.apache.commons.io.FileUtils;
+import org.apache.tika.Tika;
+import org.apache.tika.mime.MediaType;
+import org.apache.tika.mime.MediaTypeRegistry;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ScanTypeDecider {
+    private final Logger logger = LoggerFactory.getLogger(this.getClass());
+    
+    public Map<DetectTool, Set<String>> decide(boolean hasImageOrTar, DetectPropertyConfiguration detectConfiguration, Path detectSourcePath) {
+        if (!hasImageOrTar && detectConfiguration.getValue(DetectProperties.DETECT_AUTONOMOUS_SCAN_ENABLED)) {
+            AllNoneEnumCollection<DetectTool> includedTools = detectConfiguration.getValue(DetectProperties.DETECT_TOOLS);
+            AllNoneEnumCollection<DetectTool> excludedTools = detectConfiguration.getValue(DetectProperties.DETECT_TOOLS_EXCLUDED);
+            if (detectSourcePath == null) {
+                logger.error("Detect autonomous scan mode requires Detect Source Path (--detect.source.path) to be set.");
+            } else {
+                // This map has individual paths to files grouped under types.
+                PathsCollection pathsCollection = search(detectSourcePath);
+                logger.debug("includedTools: {}", includedTools.toPresentValues());
+                logger.debug("excludedTools: {}", excludedTools.toPresentValues());
+                final Map<DetectTool, Set<String>> scanTypeEvidenceMap = new HashMap<>();
+                if (!excludedTools.containsValue(DetectTool.BINARY_SCAN)
```

Comment:
Maybe add a private `decideTool()` method and call it for each tool instead of duplicating the if statements?

Reply:
Done, thanks!